src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -67,6 +70,14 @@ var ReactDOMInput = {
         props,
         inst._currentElement._owner
       );
+
+      if (props.valueLink !== undefined && !didWarnValueLink) {
+        warning(
+          false,
+          '`.valueLink` on `input` is deprecated; set `.value` and `.onChange` instead.'
```

Comment:
nit: can we not have dots before all of these?

nit: checkedLink too?

nit: might be simpler to put this logic in LinkedValueUtils instead of in each wrapper.